streamz/sources.py

```diff
@@ -237,22 +237,75 @@ def _close_consumer(self):
         self.stopped = True
 
 
+class ConsumerFactory:
+    def __init__(self, consumer_params, topic):
+        self.consumer_params = self._fix_params(consumer_params)
+        self.topic = topic
+        self.consumers = {}
+        self.ck = None
+
+    def _fix_params(self, consumer_params):
+        if 'group.id' not in consumer_params:
+            import warnings
+            import uuid
+            group_id = "consumer-{}".format(uuid.uuid4())
+            warnings.warn("Consumer group id is not specified, Job restarts consume messages from offset 0.\n" +
+                          "Use group.id: {} for restarting this job from a check point in future.".format(group_id))
+            consumer_params['group.id'] = group_id
+
+        consumer_params['enable.auto.commit'] = 'false'
```

Comment:
What was the reason behind this and the next (commented) line?

Reply:
I am manually committing the offsets after each batch is read successfully. And the job would restart from this checkpoint and not from an auto-commit in case of a failure. 

This configuration specifies the maximum time a consumer can stay idle between two `poll` operations. A consumer might stay idle for a longer period with current implementation. For example if Dask cluster is still processing older batches and the consumer is waiting for a thread to poll next batch.